packages/core/src/services/claimedRewards/claim.ts

```diff
@@ -0,0 +1,69 @@
+import { REWARD_VALUES, RewardType, User, Workspace } from '../../browser'
+import { database } from '../../client'
+import { publisher } from '../../events/publisher'
+import { ForbiddenError, Result, Transaction } from '../../lib'
+import { ClaimedRewardsRepository } from '../../repositories'
+import { claimedRewards } from '../../schema'
+
+export async function claimReward(
+  {
+    workspace,
+    user,
+    type,
+    reference,
+  }: {
+    workspace: Workspace
+    user: User
+    type: RewardType
+    reference: string
+  },
+  db = database,
+) {
+  if (type === RewardType.Referral) {
+    publisher.publishLater({
+      type: 'sendReferralInvitation',
+      data: {
+        email: reference,
+        workspaceId: workspace.id,
+        userId: user.id,
+      },
+    })
+  }
```

Comment:
the referral should be sent even if the reward fails to get created?

Reply:
hhmm nope, fixed it!